### PR TITLE
Restrict 4.5.1 to TLS 1.2, and centralize version choice

### PR DESCRIPTION
**Pull request by drwill-ms**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [x] This pull-request is submitted against the `master` branch.

## Description of the changes
Retarget .NET Framework 4.5.1 code to only specify TLS 1.2 (removing 1.0 and 1.1). It does not have a SystemDefault setting, which would be preferred, so we must be explicit. Note this is one of many reasons why 4.5.1 support should be deprecated.

## Reference/Link to the issue solved with this PR (if any)
6024224

### CODE REVIEW

File: iothub/device/src/InternalClient.cs [hunk 1]
```diff
@@ -111,9 +110,7 @@ public InternalClient(IotHubConnectionString iotHubConnectionString, ITransportS
         {
             if (Logging.IsEnabled) Logging.Enter(this, transportSettings, pipelineBuilder, nameof(InternalClient) + "_ctor");
 
-#if NET451
-            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls;
-#endif
+            TlsVersions.SetLegacyAcceptableVersions();
```
Comment (prmathur-microsoft) [inline iothub/device/src/InternalClient.cs]: Does this take care of restricting in amqp/mqtt as well ?

Comment (drwill-ms) [reply iothub/device/src/InternalClient.cs]: From my testing, it only seems to impact HTTP or MQTT/AMQP over websocket. Based on documentation and of the frameworks we support, it should impact .NET 4.5.1 only.

My next PR should address straight up MQTT.

File: tools/TLS Protocols Tests/TLS protocol tests.csproj [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>netcoreapp2.0;net451</TargetFrameworks>
```
Comment (abhipsaMisra) [inline tools/TLS Protocols Tests/TLS protocol tests.csproj]: csproj update for linux: https://github.com/drwill-ms/azure-iot-sdk-csharp/pull/2

File: provisioning/transport/mqtt/src/ProvisioningTransportHandlerMqtt.cs [hunk 3]
```diff
@@ -179,8 +179,11 @@ private static DeviceRegistrationResult ConvertToProvisioningRegistrationResult(
                 ((SecurityProviderX509)message.Security).GetAuthenticationCertificate();
 
             var tlsSettings = new ClientTlsSettings(
-                message.GlobalDeviceEndpoint,
-                new List<X509Certificate> { clientCertificate });
+                TlsVersions.AcceptableVersions,
```
Comment (drwill-ms) [inline provisioning/transport/mqtt/src/ProvisioningTransportHandlerMqtt.cs]: confirm this fails if set to none in .net core app

File: tools/TLS Protocols Tests/Program.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        /// <summary>
+        /// Validates that the registry manager, and all the device client transports will succeed or fail given the OS settings configured.
+        /// </summary>
+        /// <remarks>
+        /// To observe messages from the hub, consider using the az cli:
+        ///   az iot hub monitor-events -n "your hub name"
+        /// </remarks>
+        static async Task Main(string[] args)
+        {
+            string hubConnectionString = Environment.GetEnvironmentVariable("HubConnectionString");
```
Comment (drwill-ms) [inline tools/TLS Protocols Tests/Program.cs]: Make these the same as E2E env vars

File: common/test/Configuration.IoTHub.cs [hunk 5]
```diff
@@ -25,12 +25,6 @@ public static X509Certificate2 GetCertificateWithPrivateKey()
 
             public static string DeviceConnectionStringInvalidServiceCertificate => GetValue("IOTHUB_DEVICE_CONN_STRING_INVALIDCERT", string.Empty);
 
-            public static string DeviceConnectionString => GetValue("IOTHUB_DEVICE_CONN_STRING");
```
Comment (drwill-ms) [inline common/test/Configuration.IoTHub.cs]: See if these values are in the cmd

File: e2e/test/netstandard20/ProvisioningE2ETests.cs [hunk 6]
```diff
@@ -384,7 +384,7 @@ private async Task ProvisioningDeviceClient_CustomAllocationPolicy(Client.Transp
             await ProvisioningDeviceClient_ProvisioningFlow_CustomAllocation_AllocateToHubWithLongestHostName(transportProtocol, attestationType, enrollmentType, setCustomProxy, iotHubsToProvisionTo, expectedDestinationHub, customServerProxy).ConfigureAwait(false);
         }
 
-#endregion
+        #endregion
```
Comment (drwill-ms) [inline e2e/test/netstandard20/ProvisioningE2ETests.cs]: space?

File: common/test/Configuration.IoTHub.cs (hunk 5 above)
Comment (drwill-ms) [reply common/test/Configuration.IoTHub.cs]: Used in samples and stress tests, so leaving for convenience.

File: e2e/test/netstandard20/ProvisioningE2ETests.cs (hunk 6 above)
Comment (drwill-ms) [reply e2e/test/netstandard20/ProvisioningE2ETests.cs]: Matches open region

**Pull request merged**